### PR TITLE
Guard against missing currentGallery element

### PR DESCRIPTION
**Pull request by rgalanakis**
When an image is loaded,
it can preload the next/previous image with a callback.
It's possible that between when the decision to preload is made
and the `loadImage` call for preloading happens,
the current gallery has been modified and the index to preload
is missing, causing the preload callback to error.

---

I can't say _exactly_ why this is occurring, but each day we get at least one error on this line, with a stacktrace coming from `preloadNext` or `preloadPrev`, and this change appears to fix it. I decided to not invoke the callback in the case the chosen item is missing from the current gallery, since that's what you do on line 474 (if `imagesElements[index]` is undefined). But since I can't create a reliable repro, I could use your judgement here.

### CODE REVIEW

**Comment by feimosi** [comment]
Thanks for reporting that. I'm currently unavailable, so I'm gonna need more time to investigate it.

File: src/baguetteBox.js [hunk 1]
```diff
@@ -484,8 +484,15 @@
             return;
         }
 
+        var galleryItem = currentGallery[index];
+        // It's possible that the decision to preload is made,
+        // but by the time the load happens, the element is gone.
+        if (typeof galleryItem  === 'undefined') {
+            return;
+        }
```
Comment (feimosi) [inline src/baguetteBox.js]: I'd rename the variable to `galleryImage` and move the whole to the top to become:
```js
var imageContainer = imagesElements[index];
var galleryImage = currentGallery[index];

// Return if the index exceeds prepared images in the overlay
// or if the current gallery has been changed / closed
if (imageContainer === undefined || galleryImage === undefined) {
    return;
}
```
I've also updated the comment. Let me know what do you think.

**Comment by feimosi** [comment]
Your reasoning makes perfect sense and that may be the case. See my review and thanks for the contribution!

**Comment by rgalanakis** [comment]
Ok, updated.

**Comment by feimosi** [comment]
Perfect, thanks!

**Pull request merged**